docs/support_files/hooks.md

```diff
@@ -102,3 +102,19 @@ AfterAll(function () {
   return Promise.resolve()
 });
 ```
+
+# BeforeStep / AfterStep
+
+If you have some code execution that needs to be done before or after all steps, use `BeforeStep` / `AfterStep`. Like the `Before` / `After` hooks, these also have a world instance as 'this', and can be conditionally selected for execution based on the tags of the scenario.
+
+```javascript
+var {AfterStep, BeforeStep} = require('cucumber');
+
+BeforeStep(function () {
+  // This hook will be executed before all steps
+});
+
+AfterStep({tags: "@foo"}, function () {
```

Comment:
I think it would be good in this example or another to show the function consuming the step result, as users will want to know how to do this.